doc/design/parallel-node-config.md

```diff
@@ -0,0 +1,216 @@
+* A node could be included into several pools. In this case pool should be 
+  selected by it’s priority like it’s implemented for SriovNetworkNodePolicy
+* Move all drain-related logic into the centralized place
+
+
+### Non-Goals
+Parallel NICs configuration on the same node is out of scope of this proposal
+
+## Proposal
+Introduce nodes pool drain configuration and controller to meet Goals targets.
+
+
+### Workflow Description
+A new Drain controller will be introduced to manage node drain and cordon procedures. That means we don't need to do
+drain and use `drain lock` in config daemon anymore. The overall drain process will be covered by the following states:
+
+```golang
+NodeDrainAnnotation             = "sriovnetwork.openshift.io/state"
+NodeStateDrainAnnotation        = "sriovnetwork.openshift.io/desired-state"
+NodeStateDrainAnnotationCurrent = "sriovnetwork.openshift.io/current-state"
+DrainIdle                       = "Idle"
+DrainRequired                   = "Drain_Required"
+RebootRequired                  = "Reboot_Required"
+Draining                        = "Draining"
+DrainComplete                   = "DrainComplete"
+```
+
+Drain controller will watch for Node annotation, `sriovnetwork.openshift.io/state` 
+and SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state`
+and write the `sriovnetwork.openshift.io/current-state` annotation in the SriovNetworkNodeState.
+
+Config Daemon will read SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state` and write both
+Node annotation, `sriovnetwork.openshift.io/state` and SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state`.
+
+*NOTE:* In the future we are going to drop the node annotation and only use the SriovNetworkNodeState
+
+Draining procedure:
+
+1. config daemon mark the node as `Drain_Required` or `Reboot_Required` by adding that to both the Node annotation, `sriovnetwork.openshift.io/state`
+   and SriovNetworkNodeState annotation `sriovnetwork.openshift.io/desired-state`
+2. operator drain controller reconcile loop find the node SriovNetworkPoolConfig
+   1. if number of `Draining` nodes is great or equal to the `MaxUnavailable` the operator will re-queue the request
+   2. if number of `Draining` nodes is lower than the `MaxUnavailable` the operator will start the draining process
+   and annotate the SriovNetworkNodeState annotation `sriovnetwork.openshift.io/current-state` with `Draining`
+5. on Openshift platform we will pause the machine config pool related to the node
+6. the operator will start the drain process
+   1. if `Drain_Required` the operator will remove ONLY pods used sriov devices
+   2. if `Reboot_Required` the operator will remove ALL the pods on the system
+9. operator moves the `sriovnetwork.openshift.io/current-state` annotation to `DrainComplete`
+10. daemon will continue to the configuration when it's done it will move back the `sriovnetwork.openshift.io/state` annotation to `Idle`
```

Comment:
you mean both `sriovnetwork.openshift.io/state` on node and `sriovnetwork.openshift.io/desired-state` on SriovNetworkNodeState